### ISSUE TITLE
Database and repository incident

### ISSUE THREAD

**Issue by thisisusernameforgit** [go-gitea/gitea#12580]
- Gitea version (or commit ref): 1.12.3
- Git version: 2.22.0
- Operating system: Centos 7
- Database (use `[x]`):
  - [x] PostgreSQL
- Can you reproduce the bug at https://try.gitea.io:
  - [x] No

## Description
I previously created issue #12570 

since could not solve the problem i had to restore the backup from August 21st.
what will happen and how to solve it (if this is a problem), if the database returned to the state on August 21, and changes were made to the repository files during the entire downtime of the Gitea service (August 21-22-23), new files were uploaded, etc. etc.

1. What should I expect, what errors may occur?
2. Is it worth running a forced scan of repositories or something similar?
    2.1. if I have to do something, what exactly do I need to do

**Comment by fake-name** [comment]
This is a super old issue, but I recently went through the same issues (migrating from gogs to gitea).

I got past https://github.com/go-gitea/gitea/issues/12570 by turning on query logging, and looking at the queries the gitea app was issuing. In this case, there's a column (`index`) missing from the `repo_unit` table. Manually adding it (`ALTER TABLE repo_unit ADD COLUMN index BIGINT;`) lets the migrations run to current, and gitea will then successfully start.

Other broken stuff:

 - Git auth is broken. Fix by `ALTER TABLE repository ADD COLUMN is_bare BOOLEAN DEFAULT true`
 - The actual repos seem to not come over.

Some caveats:
 - There's LOTS of warnings about `Table xxx has column xxx but struct has not related field`.

Basically, you shouldn't do this, I think. I'm having fun breaking things, though!
